frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -45,58 +45,52 @@ local function cleanupSelectedText(text)
 end
 
 function ReaderHighlight:init()
+    self.select_mode = false -- extended highlighting
+
     self._highlight_buttons = {
-        -- highlight and add_note are for the document itself,
-        -- so we put them first.
-        ["01_highlight"] = function(_self)
+        ["01_select"] = function(_self)
             return {
-                text = _("Highlight"),
+                text = _("Select"),
+                enabled = _self.hold_pos ~= nil,
                 callback = function()
-                    _self:saveHighlight()
+                    _self:startSelection()
                     _self:onClose()
                 end,
-                enabled = _self.hold_pos ~= nil,
             }
         end,
-        ["02_add_note"] = function(_self)
+        ["02_highlight"] = function(_self)
             return {
-                text = _("Add Note"),
+                text = _("Highlight"),
                 callback = function()
-                    _self:addNote()
+                    _self:saveHighlight()
                     _self:onClose()
                 end,
                 enabled = _self.hold_pos ~= nil,
             }
         end,
-        -- copy and search are internal functions that don't depend on anything,
-        -- hence the second line.
         ["03_copy"] = function(_self)
             return {
                 text = C_("Text", "Copy"),
                 enabled = Device:hasClipboard(),
                 callback = function()
                     Device.input.setClipboardText(cleanupSelectedText(_self.selected_text.text))
                     _self:onClose()
-                    self:clear()
                     UIManager:show(Notification:new{
                         text = _("Selection copied to clipboard."),
                     })
                 end,
             }
         end,
-        ["04_search"] = function(_self)
+        ["04_add_note"] = function(_self)
             return {
-                text = _("Search"),
+                text = _("Add Note"),
                 callback = function()
-                    _self:onHighlightSearch()
-                    -- We don't call _self:onClose(), crengine will highlight
-                    -- search matches on the current page, and self:clear()
-                    -- would redraw and remove crengine native highlights
+                    _self:addNote()
+                    _self:onClose()
                 end,
+                enabled = _self.hold_pos ~= nil,
             }
         end,
-        -- then information lookup functions, putting on the left those that
-        -- depend on an internet connection.
```

Comment:
I know you don't like comments - but does that really go to the point of deleting exisitng ones ? :)

There we some kind of logic before that these comments describe. We didn't change much of it, so keep the ones that are still valid. And add one explaining why Search is at the end (some new logic we added, let it be explicite in the code so you or me or our successors knows about it next time we decide to change the ordering).